### ISSUE TITLE
Line numbers in stacktraces incorrect, not matching with actual file numbers

### ISSUE THREAD

**Issue by corneliusroemer** [snakemake/snakemake#1247]
Snakemake version: 6.9.1
Host OS: CentOS
Submission via: SLURM
Repo where bug occurs: https://github.com/neherlab/ncov-simple/tree/5bdbdf3df5c3e5833fccff4fe35a487d24313ff6

**Describe the bug**
The line numbers in stacktraces sometimes don't correspond to the actual line numbers in the files.

Stacktrace says bug in line 680:

![image](https://user-images.githubusercontent.com/25161793/139855554-7666824b-4ba4-4aee-9ab8-eb57c62001c5.png)

But the file is only 570 lines long:

![image](https://user-images.githubusercontent.com/25161793/139855809-ea1762e3-a666-47ed-a122-210054b665b1.png)

Here's where the rule actually is, in line 350 something:

![image](https://user-images.githubusercontent.com/25161793/139855815-3e422df5-ec6d-4102-82de-8fa65ce0a2e3.png)

**Additional context**
In the [Snakefile](https://github.com/neherlab/ncov-simple/blob/5bdbdf3df5c3e5833fccff4fe35a487d24313ff6/Snakefile), we're importing this `core.smk` file, could this be relevant for this bug?
```
if len(config["builds"]):
    include: "workflow/snakemake_rules/subsampling.smk"
    include: "workflow/snakemake_rules/core.smk"
```

**Comment by corneliusroemer** [comment]
This is so weird and quite problematic when debugging.

In snakemake 6.9.1 it's borken

In snakemake 6.8.0 no bug, so seems like a regression?

**Comment by corneliusroemer** [comment]
This is a serious bug and it's still present in v6.10.0, just tested.

Can someone from the core team have a look @johanneskoester? Wrong stacktraces are pretty bad for debugging...

**Comment by daiuc** [comment]
I run into the exact same issue with v6.10.0.

**Comment by ASLeonard** [comment]
I've come across this issue in multiple contexts, easy to reproduce minimally 

- invalid access of input/output file (same as OP's report) -> Line 9, reported as Line 11
- syntax errors (see by uncommenting #a) -> Line 4, reported as Line 10
- duplicate rules (uncomment 2nd rule) -> Line 10, reported as Line 25
```
rule all:
    input:
        'test.txt'
#a #SYNTAX ERROR
rule test:
    output:
       a='test.txt'
    shell:
        'echo 1 > {output.b}' #NAME ERROR
#rule test: #DUPLICATE ERROR
#    output:
#      a='test.txt'
#    shell:
#        'echo 1 > {output.b}' 
```

**Comment by dariober** [comment]
Just found out that my issue https://github.com/snakemake/snakemake/issues/1256 is a duplicate of this one.

**Comment by dariober** [comment]
I think this has been fixed in v6.12.3.

**Comment by ASLeonard** [comment]
Yeah this appears to solve the reproducible errors above in at least v6.13.1

**Comment by corneliusroemer** [comment]
Seems to have been fixed in v6.12.3, thanks @dariober
